configure.ac

```diff
@@ -74,8 +72,14 @@ AC_CHECK_LIB(dl, dlopen, [],
 dnl Checks for libraries
 AC_CHECK_LIB(pthread, pthread_mutex_init, [],
     AC_MSG_ERROR([pthread_mutex_init() not found.  libfabric requires libpthread.]))
-AC_CHECK_LIB(rt, clock_gettime, [],
-    AC_MSG_ERROR([clock_gettime() not found.  libfabric requires librt.]))
+
+AC_SEARCH_LIBS([clock_gettime],[rt posix4],
+		[AC_DEFINE([HAVE_CLOCK_GETTIME], [1],
+			[Define to 1 if have clock_gettime.])],
+		[AS_IF([test -z $(expr "$host_os" : '.*\(darwin\)')],
+			[AC_MSG_ERROR([clock_gettime not found.])],
```

Comment:
FWIW, it's better/safer to always AC_DEFINE logical macro values to be 0 or 1 -- not undef or 1.

Rationale: if you have the following in your C code:

```c
#ifdef HAVE_CLOCK_GETTTIME
...
```

You won't get any warning from the compiler if you misspell the macro name (like it's misspelled above -- did you notice? :-) ).

But if your configury *always* AC_DEFINEs HAVE_CLOCK_GETTIME to 0 or 1, then your C code can be:

```c
#if HAVE_CLOCKGETTIME
...
```

and you'll get a warning (or possibly even an error) from the compiler if that macro was not defined.

I'm a fan of defensive coding.  :-)